### PR TITLE
api: add region label api

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve?

Add API for region label rules. Can review now but should be merged after https://github.com/tikv/pd/pull/3968
Review diff from [this commit](https://github.com/tikv/pd/pull/3984/commits/0ac942b3c77abf2f4c5f2f172396eb4138d6db63)

Fix https://github.com/tikv/pd/issues/3985

### Check List

Tests
- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/api/region_label.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+func (h *regionLabelHandler) GetAllRules(w http.ResponseWriter, r *http.Request) {
+	cluster := getCluster(r)
+	rules := cluster.GetRegionLabeler().GetAllLabelRules()
+	h.rd.JSON(w, http.StatusOK, rules)
+}
+
+// @Tags region_label
+// @Summary Update region label rules in batch.
+// @Produce json
+// @Success 200 {array} labeler.LabelRulePatch
```
Comment (rleungx) [inline server/api/region_label.go:54]: Also need 400 and 500

File: server/api/region_label.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	}
+	h.rd.JSON(w, http.StatusOK, "Update region label rules successfully.")
+}
+
+// @Tags region_label
+// @Summary Get label rules of cluster by ids.
+// @Param body body []string true "IDs of query rules"
+// @Produce json
+// @Success 200 {array} labeler.LabelRule
+// @Failure 404 {string} string "The rule does not exist."
```
Comment (rleungx) [inline server/api/region_label.go:78]: need 500

File: server/api/region_label.go (hunk 1 above)
Comment (rleungx) [reply server/api/region_label.go:54]: BTW, `patch` doesn't return an array.

**Review by rleungx — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]
LGTM. BTW, do we have any integration tests on these APIs with TiDB now?

**Pull request merged**